### PR TITLE
1-minute fetch window for ETH price

### PR DESCRIPTION
**Pull request by 0xZenus**

### CODE REVIEW

File: src/utils/priceFetch.ts [hunk 1]
```diff
@@ -1,22 +1,34 @@
...
 
-  for (let attempt = 0; attempt < attempts; attempt += 1) {
+  price = 0;
+  for (let attempt = 0; attempt < ATTEMPTS; attempt += 1) {
     try {
       const data = await axios.get(
         `https://pro-api.coingecko.com/api/v3/simple/price?x_cg_pro_api_key=${apiKey}&ids=ethereum&vs_currencies=usd`,
       );
-      return data.data.ethereum.usd;
+      price = data.data.ethereum.usd;
```
Comment (ssbarbee) [inline src/utils/priceFetch.ts:22]: Previously the code stops if we manage to fetch eth price on first try due to the return. This code will fetch the eth five times in a row even though you get the eth price on the first go.

I think you introduce a small bug.

File: src/utils/priceFetch.ts [hunk 2]
```diff
@@ -1,22 +1,34 @@
 import axios from 'axios';
 import { getSentryTracker } from '../init';
 
+const FETCH_WINDOW_IN_MS = 60 * 1000;
+const ATTEMPTS = 5;
+
+let price = 0;
+let timestampInMS = 0;
+
```
Comment (ssbarbee) [inline src/utils/priceFetch.ts:9]: Add tests for this function. Seems good candidate to test.

**Review by ssbarbee — CHANGES_REQUESTED** [review]

File: src/utils/priceFetch.ts (hunk 1 above)
Comment (0xZenus) [reply src/utils/priceFetch.ts:22]: Great spot!

File: src/utils/priceFetch.ts (hunk 2 above)
Comment (0xZenus) [reply src/utils/priceFetch.ts:9]: Yes, let me do this today and I'll update the PR.

File: tests/mocha/priceFetch.test.ts [hunk 3]
```diff
@@ -0,0 +1,23 @@
+import { expect } from 'chai';
+import * as sinon from 'sinon';
+import axios from 'axios';
+import { geckoEthToUsd } from '../../src/utils/priceFetch';
+
+describe('ETH price fetch', () => {
+  describe('fetch price 1', () => {
```
Comment (ssbarbee) [inline tests/mocha/priceFetch.test.ts:7]: add a test case when error is throw by axios. verify that you will call the stubbed function 5 times.

File: tests/mocha/priceFetch.test.ts [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+          ethereum: {
+            usd: 1000,
+          },
+        },
+      });
+
+      const price = await geckoEthToUsd('');
+      expect(price).to.be.eq(1000);
+
+      (axios.get as sinon.SinonStub).restore();
```
Comment (ssbarbee) [inline tests/mocha/priceFetch.test.ts:20]: verify that you are calling the stubbed function with proper parameters.

**Pull request closed without merging**